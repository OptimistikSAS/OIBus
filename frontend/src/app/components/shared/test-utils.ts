import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
