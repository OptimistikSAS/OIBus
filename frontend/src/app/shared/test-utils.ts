import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';
import { Page } from './types';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };

export function toPage<T>(content: Array<T>, totalElements: number = content.length, num = 0, size = Math.max(20, totalElements)): Page<T> {
  return {
    content,
    totalElements,
    number: num,
    size,
    totalPages: Math.ceil(totalElements / size)
  };
}

export function emptyPage<T>(size = 20): Page<T> {
  return {
    totalElements: 0,
    content: [] as Array<T>,
    size,
    totalPages: 0,
    number: 0
  };
}
