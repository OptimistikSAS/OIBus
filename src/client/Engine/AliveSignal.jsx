import React from 'react'
import PropTypes from 'prop-types'
import { Col, Row } from 'reactstrap'
import { OIbText, OIbTitle, OIbCheckBox, OIbInteger, OIbProxy, OIbAuthentication } from '../components/OIbForm'
import validation from './Engine.validation'

const AliveSignal = ({ onChange, aliveSignal }) => (
  <>
    <OIbTitle label="AliveSignal parameters">
      <>
        <p>This component allows to send an alive signal on a given frequency to the chosen host.</p>
      </>
    </OIbTitle>
    <Row>
      <Col md={3}>
        <OIbCheckBox
          label="Enabled"
          name="engine.aliveSignal.enabled"
          value={aliveSignal.enabled}
          defaultValue={false}
          onChange={onChange}
          switchButton
        />
      </Col>
    </Row>
    {aliveSignal.enabled && (
      <Row>
        <Col md={2}>
          <OIbText
            label="Host"
            name="engine.aliveSignal.host"
            value={aliveSignal.host}
            defaultValue=""
            valid={validation.engine.aliveSignal.host}
            help={<div>The host to send the alive signal</div>}
            onChange={onChange}
          />
        </Col>
        <Col md={2}>
          <OIbText
            label="Endpoint"
            name="engine.aliveSignal.endpoint"
            value={aliveSignal.endpoint}
            defaultValue=""
            valid={validation.engine.aliveSignal.endpoint}
            help={<div>The endpoint send the alive signal</div>}
            onChange={onChange}
          />
        </Col>
        <Col md={2}>
          <OIbText
            label="Id"
            name="engine.aliveSignal.id"
            value={aliveSignal.id}
            defaultValue="OIBus"
            valid={validation.engine.aliveSignal.id}
            help={<div>Message id</div>}
            onChange={onChange}
          />
        </Col>
        <Col md={2}>
          <OIbInteger
            label="Frequency"
            name="engine.aliveSignal.frequency"
            value={aliveSignal.frequency}
            defaultValue={600}
            valid={validation.engine.aliveSignal.frequency}
            help={<div>How often to send the alive signal in seconds</div>}
            onChange={onChange}
          />
        </Col>
        <Col md={2}>
          <OIbProxy
            label="Proxy"
            name="engine.aliveSignal.proxy"
            value={aliveSignal.proxy}
            defaulValue
            help={<div>Proxy name to use</div>}
            onChange={onChange}
          />
        </Col>
      </Row>
    )}
    {aliveSignal.enabled && (
      <Row>
        <Col md={8}>
          <OIbAuthentication
            label="AliveSignal authentication parameters"
            name="engine.aliveSignal.authentication"
            value={aliveSignal.authentication}
            defaulValue
            onChange={onChange}
          />
        </Col>
      </Row>
    )}
  </>
)
AliveSignal.propTypes = {
  onChange: PropTypes.func.isRequired,
  aliveSignal: PropTypes.object.isRequired,
}

export default AliveSignal
