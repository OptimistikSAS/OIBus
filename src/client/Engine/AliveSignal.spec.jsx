import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'

import testConfig from '../../../tests/testConfig'
import AliveSignal from './AliveSignal.jsx'

const mockMath = Object.create(global.Math)
mockMath.random = () => 1
global.Math = mockMath

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

describe('AliveSignal', () => {
  test('check AliveSignal', () => {
    act(() => {
      ReactDOM.render(<AliveSignal
        aliveSignal={testConfig.engine.aliveSignal}
        onChange={() => (1)}
      />, container)
    })
    expect(container).toMatchSnapshot()
  })
})
