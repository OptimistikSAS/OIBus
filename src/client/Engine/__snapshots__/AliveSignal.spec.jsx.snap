// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AliveSignal check AliveSignal 1`] = `
<div>
  <div
    class="row"
  >
    <h5>
      AliveSignal parameters
      <button
        aria-label="Close"
        class="close"
        id="id"
      >
        <svg
          class="oi-help"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
          />
        </svg>
      </button>
    </h5>
  </div>
  <div
    class="container-fluid"
  >
    <div
      class="collapse"
      toggler="id"
    >
      <div
        class="row"
        style="margin-bottom: 15px;"
      >
        <div
          class="col"
        >
          <p>
            This component allows to send an alive signal on a given frequency to the chosen host.
          </p>
          <p>
            The chosen host can also be an another OIBus instance which will forward the message based on its own AliveSignal configuration.
          </p>
          <p>
            You can check the number of forwarded messages on the OIBus health status screen looking for the forwardedAliveSignalMessages field.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-md-3"
    >
      <div
        class="form-group"
      >
        <div
          class="custom-switch custom-control"
        >
          <input
            checked=""
            class="custom-control-input"
            color="secondary"
            id="engine.aliveSignal.enabled"
            name="engine.aliveSignal.enabled"
            type="checkbox"
          />
          <label
            class="custom-control-label"
            for="engine.aliveSignal.enabled"
          >
            Enabled
          </label>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-md-2"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="engine.aliveSignal.host"
        >
          Host
        </label>
        <input
          aria-invalid="false"
          class="oi-form-input form-control"
          id="engine.aliveSignal.host"
          name="engine.aliveSignal.host"
          type="text"
          value="https://demo.host"
        />
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          <div>
            The host to send the alive signal
          </div>
        </small>
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="engine.aliveSignal.endpoint"
        >
          Endpoint
        </label>
        <input
          aria-invalid="false"
          class="oi-form-input form-control"
          id="engine.aliveSignal.endpoint"
          name="engine.aliveSignal.endpoint"
          type="text"
          value="/api/optimistik/oibus/info"
        />
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          <div>
            The endpoint send the alive signal
          </div>
        </small>
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="engine.aliveSignal.id"
        >
          Id
        </label>
        <input
          aria-invalid="false"
          class="oi-form-input form-control"
          id="engine.aliveSignal.id"
          name="engine.aliveSignal.id"
          type="text"
          value="OIBus-test"
        />
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          <div>
            Message id
          </div>
        </small>
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="engine.aliveSignal.frequency"
        >
          Frequency
        </label>
        <input
          aria-invalid="false"
          class="oi-form-input form-control"
          id="engine.aliveSignal.frequency"
          name="engine.aliveSignal.frequency"
          type="integer"
          value="300"
        />
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          <div>
            How often to send the alive signal in seconds
          </div>
        </small>
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="engine.aliveSignal.proxy"
        >
          Proxy
        </label>
        <select
          class="oi-form-input form-control"
          id="engine.aliveSignal.proxy"
          name="engine.aliveSignal.proxy"
        >
          <option
            value=""
          />
        </select>
        <small
          class="form-text text-muted"
        >
          <div>
            Proxy name to use
          </div>
        </small>
      </div>
    </div>
    <div
      class="col-md-2"
    />
  </div>
  <div
    class="row"
  >
    <div
      class="col-md-8"
    >
      <div
        class="row"
      >
        <h5>
          AliveSignal authentication parameters
          <button
            aria-label="Close"
            class="close"
            id="id"
          >
            <svg
              class="oi-help"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
              />
            </svg>
          </button>
        </h5>
      </div>
      <div
        class="container-fluid"
      >
        <div
          class="collapse"
          toggler="id"
        >
          <div
            class="row"
            style="margin-bottom: 15px;"
          >
            <div
              class="col"
            >
              <div>
                <p>
                  Authentication parameters
                </p>
                <p>
                  Please fill the user and password to connect this application
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-2"
        >
          <div
            class="form-group"
          >
            <label
              class=""
              for="type"
            >
              Type
            </label>
            <select
              aria-invalid="false"
              class="oi-form-input form-control"
              id="type"
              name="type"
            >
              <option
                value="Basic"
              >
                Basic
              </option>
            </select>
            <div
              class="invalid-feedback"
            />
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-4"
        >
          <div
            class="form-group"
          >
            <label
              class=""
              for="username"
            >
              User name
            </label>
            <input
              aria-invalid="false"
              class="oi-form-input form-control"
              id="username"
              name="username"
              type="text"
              value="username"
            />
            <div
              class="invalid-feedback"
            />
          </div>
        </div>
        <div
          class="col-md-4"
        >
          <div
            class="form-group"
          >
            <label
              class=""
              for="password"
            >
              Password
            </label>
            <div
              class="input-group"
            >
              <input
                aria-invalid="false"
                class="oi-form-input form-control"
                id="password"
                name="password"
                type="password"
                value="password"
              />
            </div>
            <div
              class="invalid-feedback"
              style="display: inline;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`AliveSignal check change aliveSignal to false 1`] = `
<div>
  <div
    class="row"
  >
    <h5>
      AliveSignal parameters
      <button
        aria-label="Close"
        class="close"
        id="id"
      >
        <svg
          class="oi-help"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
          />
        </svg>
      </button>
    </h5>
  </div>
  <div
    class="container-fluid"
  >
    <div
      class="collapse"
      toggler="id"
    >
      <div
        class="row"
        style="margin-bottom: 15px;"
      >
        <div
          class="col"
        >
          <p>
            This component allows to send an alive signal on a given frequency to the chosen host.
          </p>
          <p>
            The chosen host can also be an another OIBus instance which will forward the message based on its own AliveSignal configuration.
          </p>
          <p>
            You can check the number of forwarded messages on the OIBus health status screen looking for the forwardedAliveSignalMessages field.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-md-3"
    >
      <div
        class="form-group"
      >
        <div
          class="custom-switch custom-control"
        >
          <input
            checked=""
            class="custom-control-input"
            color="secondary"
            id="engine.aliveSignal.enabled"
            name="engine.aliveSignal.enabled"
            type="checkbox"
          />
          <label
            class="custom-control-label"
            for="engine.aliveSignal.enabled"
          >
            Enabled
          </label>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-md-2"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="engine.aliveSignal.host"
        >
          Host
        </label>
        <input
          aria-invalid="false"
          class="oi-form-input form-control"
          id="engine.aliveSignal.host"
          name="engine.aliveSignal.host"
          type="text"
          value="https://demo.host"
        />
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          <div>
            The host to send the alive signal
          </div>
        </small>
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="engine.aliveSignal.endpoint"
        >
          Endpoint
        </label>
        <input
          aria-invalid="false"
          class="oi-form-input form-control"
          id="engine.aliveSignal.endpoint"
          name="engine.aliveSignal.endpoint"
          type="text"
          value="/api/optimistik/oibus/info"
        />
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          <div>
            The endpoint send the alive signal
          </div>
        </small>
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="engine.aliveSignal.id"
        >
          Id
        </label>
        <input
          aria-invalid="false"
          class="oi-form-input form-control"
          id="engine.aliveSignal.id"
          name="engine.aliveSignal.id"
          type="text"
          value="OIBus-test"
        />
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          <div>
            Message id
          </div>
        </small>
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="engine.aliveSignal.frequency"
        >
          Frequency
        </label>
        <input
          aria-invalid="false"
          class="oi-form-input form-control"
          id="engine.aliveSignal.frequency"
          name="engine.aliveSignal.frequency"
          type="integer"
          value="300"
        />
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          <div>
            How often to send the alive signal in seconds
          </div>
        </small>
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="engine.aliveSignal.proxy"
        >
          Proxy
        </label>
        <select
          class="oi-form-input form-control"
          id="engine.aliveSignal.proxy"
          name="engine.aliveSignal.proxy"
        >
          <option
            value=""
          />
        </select>
        <small
          class="form-text text-muted"
        >
          <div>
            Proxy name to use
          </div>
        </small>
      </div>
    </div>
    <div
      class="col-md-2"
    />
  </div>
  <div
    class="row"
  >
    <div
      class="col-md-8"
    >
      <div
        class="row"
      >
        <h5>
          AliveSignal authentication parameters
          <button
            aria-label="Close"
            class="close"
            id="id"
          >
            <svg
              class="oi-help"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
              />
            </svg>
          </button>
        </h5>
      </div>
      <div
        class="container-fluid"
      >
        <div
          class="collapse"
          toggler="id"
        >
          <div
            class="row"
            style="margin-bottom: 15px;"
          >
            <div
              class="col"
            >
              <div>
                <p>
                  Authentication parameters
                </p>
                <p>
                  Please fill the user and password to connect this application
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-2"
        >
          <div
            class="form-group"
          >
            <label
              class=""
              for="type"
            >
              Type
            </label>
            <select
              aria-invalid="false"
              class="oi-form-input form-control"
              id="type"
              name="type"
            >
              <option
                value="Basic"
              >
                Basic
              </option>
            </select>
            <div
              class="invalid-feedback"
            />
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-4"
        >
          <div
            class="form-group"
          >
            <label
              class=""
              for="username"
            >
              User name
            </label>
            <input
              aria-invalid="false"
              class="oi-form-input form-control"
              id="username"
              name="username"
              type="text"
              value="username"
            />
            <div
              class="invalid-feedback"
            />
          </div>
        </div>
        <div
          class="col-md-4"
        >
          <div
            class="form-group"
          >
            <label
              class=""
              for="password"
            >
              Password
            </label>
            <div
              class="input-group"
            >
              <input
                aria-invalid="false"
                class="oi-form-input form-control"
                id="password"
                name="password"
                type="password"
                value="password"
              />
            </div>
            <div
              class="invalid-feedback"
              style="display: inline;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
